core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -194,12 +195,29 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv, numUsableCores: Int) exte
     endpoints.containsKey(name)
   }
 
-  /** Thread pool used for dispatching messages. */
-  private val threadpool: ThreadPoolExecutor = {
+  private def getNumOfThreads(conf: SparkConf): Int = {
     val availableCores =
       if (numUsableCores > 0) numUsableCores else Runtime.getRuntime.availableProcessors()
-    val numThreads = nettyEnv.conf.get(RPC_NETTY_DISPATCHER_NUM_THREADS)
+    // module configuration
+    val modNumThreads = conf.get(RPC_NETTY_DISPATCHER_NUM_THREADS)
       .getOrElse(math.max(2, availableCores))
+    // get right role
+    val executorId = conf.get(EXECUTOR_ID).getOrElse("")
```

Comment:
```
conf.get(EXECUTOR_ID).map { id =>
  val role = if (id == driver id) ...
  conf.getInt(...)
}.getOrElse(modNumThreads)
```

Cleaner. Same idea in the other similar code later.

Reply:
@vanzin , thanks for the clean code. code is rewritten. please help review.